Automattic/wp-calypso
# Domain redirects: Replace redux connect with useQuery

Follow up on https://github.com/Automattic/wp-calypso/pull/80217

Fixes https://github.com/Automattic/wp-calypso/issues/80479
Fixes https://github.com/Automattic/wp-calypso/issues/80669

## Proposed Changes

* Moves state/domains to data/domains using useQuery instead of defining all the specific redux actions
* Inlines the URL from client/lib/url/support.js, we're gonna set this support doc up right?
* s/Update/Save button

## Testing Instructions

* http://calypso.localhost:3000/domains/manage/all/test1261815.blog/edit/test1261815.word\press.com
* Redirect card should work as before:
![Screenshot 2023-08-11 at 16-23-41 test1261815 blog — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/66734dbd-f638-4e91-8df7-9aa9172cc4ee)
